$(DefineConstants) is based off of $(TargetFramework) and may be inconsistent with actual referenced assemblies

### Apple platform

iOS

### Framework version

net9.0-*

### Affected platform version

.NET 10

### Description

The set of `#define`s present within `$(DefineConstants)` is consistent with `$(TargetFramework)`, and may be *inconsistent* with the *actual* assembly used to build the project.

### Steps to Reproduce

Context: dfb9a220e15d809dcab92ff3cab95723355ca3ee

Commit dfb9a220 updated `WKUIDelegate.RunJavaScriptTextInputPanel()` to be obsolete.  However, this obsoletion was only visible on iOS 26.0 assemblies, not on iOS 18.0 assemblies; in iOS 18.0, this method is *not* obsolete.

When building an "unversioned" target framework, the *latest* reference assembly is used, which may differ from what was actually intended.

Create a new `ioslib` project:

```dotnetcli
dotnet new ioslib -n myioslib
```

Update `$(TargetFramework)`=net9.0-ios18.0 ("fixed"):

```diff
diff --git a/myioslib.csproj b/myioslib.csproj
index c62adc2..0853df5 100644
--- a/myioslib.csproj
+++ b/myioslib.csproj
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net10.0-ios</TargetFramework>
+    <TargetFramework>net9.0-ios18.0</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>true</ImplicitUsings>
```

Add the following `MyWKUIDelegate` type to the project:

```diff
diff --git a/Class1.cs b/Class1.cs
index 648114f..5f8dbae 100644
--- a/Class1.cs
+++ b/Class1.cs
@@ -1,4 +1,14 @@
+using WebKit;
+
 namespace myioslib;
 
-public class Class1 {
+public class Class1
+{
+}
+
+class MyWKUIDelegate : WKUIDelegate
+{
+    public override void RunJavaScriptTextInputPanel(WKWebView webView, string prompt, string? defaultText, WKFrameInfo frame, Action<string> completionHandler)
+    {
+    }
 }
```

Build the project:

```dotnetcli
% dotnet build -bl
```

No warnings are generated.

Update `$(TargetFramework)`=net9.0-ios ("floating"):

```diff
diff --git a/myioslib.csproj b/myioslib.csproj
index c62adc2..1e1a6ca 100644
--- a/myioslib.csproj
+++ b/myioslib.csproj
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net10.0-ios</TargetFramework>
+    <TargetFramework>net9.0-ios</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>true</ImplicitUsings>
```

Build the project:

```dotnetcli
% dotnet build -bl
Restore complete (0.7s)
    info NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  myioslib net9.0-ios succeeded with 1 warning(s) (1.5s) → bin/Debug/net9.0-ios/myioslib.dll
    …/myioslib/Class1.cs(11,26): warning CS0672: Member 'MyWKUIDelegate.RunJavaScriptTextInputPanel(WKWebView, string, string?, WKFrameInfo, Action<string>)' overrides obsolete member 'WKUIDelegate.RunJavaScriptTextInputPanel(WKWebView, string, string?, WKFrameInfo, Action<string>)'. Add the Obsolete attribute to 'MyWKUIDelegate.RunJavaScriptTextInputPanel(WKWebView, string, string?, WKFrameInfo, Action<string>)'.

Build succeeded with 1 warning(s) in 2.5s
```

Supposedly, `net9.0-ios` *should* be an alias for `net9.0-ios18.0`, but it *isn't*: net9.0-ios18.0 builds with *no* warnings, while net9.0-ios emits a warning!

Why?

If you look at the Diagnostic output from the builds, and look at the `Task "Csc"` section for `Task Parameter:` `References=`, the net9.0-ios18.0 build has:

```
$HOME/.nuget/packages/microsoft.ios.ref.net9.0_18.0/18.0.9617/ref/net9.0/Microsoft.iOS.dll
        AssemblyName=Microsoft.iOS
        AssemblyVersion=18.0.0.0
        CopyLocal=false
        ExternallyResolved=true
        FileVersion=0.0.0.0
        FrameworkReferenceName=Microsoft.iOS
        FrameworkReferenceVersion=18.0.9617
        FusionName=Microsoft.iOS, Version=18.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065
        NuGetPackageId=Microsoft.iOS.Ref.net9.0_18.0
        NuGetPackageVersion=18.0.9617
        OriginalItemSpec=$HOME/.nuget/packages/microsoft.ios.ref.net9.0_18.0/18.0.9617/ref/net9.0/Microsoft.iOS.dll
        Private=false
        PublicKeyToken=84e04ff9cfb79065
        ReferenceAssembly=$HOME/.nuget/packages/microsoft.ios.ref.net9.0_18.0/18.0.9617/ref/net9.0/Microsoft.iOS.dll
        ReferenceSourceTarget=ResolveAssemblyReference
        ResolvedFrom={RawFileName}
        Version=
```

While the net9.0-ios (floating) build has:

```
$HOME/.nuget/packages/microsoft.ios.ref.net9.0_26.0/26.0.9752/ref/net9.0/Microsoft.iOS.dll
        AssemblyName=Microsoft.iOS
        AssemblyVersion=26.0.0.0
        CopyLocal=false
        ExternallyResolved=true
        FileVersion=0.0.0.0
        FrameworkReferenceName=Microsoft.iOS
        FrameworkReferenceVersion=26.0.9752
        FusionName=Microsoft.iOS, Version=26.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065
        NuGetPackageId=Microsoft.iOS.Ref.net9.0_26.0
        NuGetPackageVersion=26.0.9752
        OriginalItemSpec=$HOME/.nuget/packages/microsoft.ios.ref.net9.0_26.0/26.0.9752/ref/net9.0/Microsoft.iOS.dll
        Private=false
        PublicKeyToken=84e04ff9cfb79065
        ReferenceAssembly=$HOME/.nuget/packages/microsoft.ios.ref.net9.0_26.0/26.0.9752/ref/net9.0/Microsoft.iOS.dll
        ReferenceSourceTarget=ResolveAssemblyReference
        ResolvedFrom={RawFileName}
        Version=
```

In particular, note that `%(AssemblyVersion)` differs: net9.0-ios18.0 has `%(AssemblyVersion)`=18.0.0.0, while net9.0-ios has `%(AssemblyVersion)`=26.0.0.0.

*Even though* net9.0-ios is supposed to be an alias for net9.0-ios18.0 for library projects, it is using the *latest* assembly!

*Furthermore*, `$(DefineConstants)` for both builds is the same!

> Task Parameter:DefineConstants=__IOS__;__MOBILE__;__UNIFIED__;TRACE;DEBUG;NET;NET9_0;NETCOREAPP;IOS;IOS18_0;NET5_0_OR_GREATER;NET6_0_OR_GREATER;NET7_0_OR_GREATER;NET8_0_OR_GREATER;NET9_0_OR_GREATER;NETCOREAPP1_0_OR_GREATER;NETCOREAPP1_1_OR_GREATER;NETCOREAPP2_0_OR_GREATER;NETCOREAPP2_1_OR_GREATER;NETCOREAPP2_2_OR_GREATER;NETCOREAPP3_0_OR_GREATER;NETCOREAPP3_1_OR_GREATER;IOS10_0_OR_GREATER;IOS10_1_OR_GREATER;IOS10_2_OR_GREATER;IOS11_0_OR_GREATER;IOS11_1_OR_GREATER;IOS11_2_OR_GREATER;IOS11_3_OR_GREATER;IOS11_4_OR_GREATER;IOS12_0_OR_GREATER;IOS12_1_OR_GREATER;IOS12_2_OR_GREATER;IOS12_3_OR_GREATER;IOS12_4_OR_GREATER;IOS13_0_OR_GREATER;IOS13_1_OR_GREATER;IOS13_2_OR_GREATER;IOS13_3_OR_GREATER;IOS13_4_OR_GREATER;IOS13_5_OR_GREATER;IOS13_6_OR_GREATER;IOS14_0_OR_GREATER;IOS14_1_OR_GREATER;IOS14_2_OR_GREATER;IOS14_3_OR_GREATER;IOS14_4_OR_GREATER;IOS14_5_OR_GREATER;IOS15_0_OR_GREATER;IOS15_2_OR_GREATER;IOS15_4_OR_GREATER;IOS16_0_OR_GREATER;IOS16_1_OR_GREATER;IOS16_2_OR_GREATER;IOS16_4_OR_GREATER;IOS17_0_OR_GREATER;IOS17_2_OR_GREATER;IOS17_4_OR_GREATER;IOS17_5_OR_GREATER;IOS18_0_OR_GREATER

There is thus no way within C# source code to know which assembly you're building against.  This is problematic if you have `$(TreatWarningsAsErrors)`=true, in which case the warning becomes an *error*.

### Did you find any workaround?

Don't use floating versions.  Always use an explicit version.

### Build logs

_No response_

--- a/docs/building-apps/build-properties.md
+++ b/docs/building-apps/build-properties.md
@@ -1325,6 +1325,35 @@ Specifies the minimum tvOS version the app can run on.
 
 Applicable to tvOS; setting this value will set [SupportedOSPlatformVersion](#supportedosplatformversion) for tvOS projects (only).
 
+## UseFloatingTargetPlatformVersion
+
+A boolean property that controls whether library projects should use a floating target platform version or the oldest available platform version.
+
+By default (starting in .NET 10), library projects without an explicit `TargetPlatformVersion` will use the oldest available reference assemblies for the current .NET version. This ensures maximum compatibility and allows library code to compile against the minimum API surface available for the target framework.
+
+However, this default behavior means that library projects are built differently than executable projects (which use the latest platform version). Code that works in an executable project may not compile when moved to a library project if it uses APIs only available in newer platform versions.
+
+Setting this property to `true` disables the automatic selection of the oldest platform version, allowing the library project to use the default (latest) platform version like executable projects do.
+
+Example:
+
+```xml
+<PropertyGroup>
+  <!-- Use the latest platform version instead of the oldest -->
+  <UseFloatingTargetPlatformVersion>true</UseFloatingTargetPlatformVersion>
+</PropertyGroup>
+```
+
+Default: `false` (use oldest platform version for library projects in .NET 10+).
+
+This property only applies to library projects (`OutputType=Library`) that are
+not app extensions and have not specified an explicit target platform version
+(the target platform version is the optional version number at the end of the
+`TargetFramework` property, for example for the TargetFramework
+`net10.0-ios26.0` the target platform version is explicitly `26.0`).
+
+This property was introduced in .NET 10.
+
 ## UseHardenedRuntime
 
 A boolean property that specifies if a hardened runtime is enabled.

--- a/scripts/generate-workloadmanifest-targets/generate-workloadmanifest-targets.cs
+++ b/scripts/generate-workloadmanifest-targets/generate-workloadmanifest-targets.cs
@@ -61,6 +61,15 @@ var highestTpvPerMajorDotNet = groupedByMajorDotNetVersion.
 				return max;
 			}).
 			ToHashSet ();
+var lowestTpvPerMajorDotNet = groupedByMajorDotNetVersion.
+			Select (gr => {
+				var min = gr.OrderBy (el => {
+					var rv = tfmToTpvAndTfv (el);
+					return float.Parse (rv.Tpv, System.Globalization.CultureInfo.InvariantCulture);
+				}).First ();
+				return min;
+			}).
+			ToHashSet ();
 
 using (var writer = new StreamWriter (outputPath)) {
 	writer.WriteLine ($"<Project>");
@@ -71,7 +80,11 @@ using (var writer = new StreamWriter (outputPath)) {
 		var tpv = parsed.Tpv;
 		supportedTFVs.Add (tfv);
 		var workloadVersion = tfm;
-		writer.WriteLine ($"	<ImportGroup Condition=\" '$(TargetPlatformIdentifier)' == '{platform}' And '$(UsingAppleNETSdk)' != 'true' And $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '{tfv}')) And '$(TargetPlatformVersion)' == '{tpv}'\">");
+		if (int.Parse (tfv.Split ('.') [0]) >= 10 && lowestTpvPerMajorDotNet.TryGetValue (tfm, out var lowest) && lowest.Split ('_') [1] == tpv) {
+			writer.WriteLine ($"	<ImportGroup Condition=\" '$(TargetPlatformIdentifier)' == '{platform}' And '$(UsingAppleNETSdk)' != 'true' And $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '{tfv}')) And ('$(TargetPlatformVersion)' == '{tpv}' Or ('$(TargetPlatformVersion)' == '' And '$(OutputType)' == 'Library' And '$(IsAppExtension)' != 'true' And '$(UseFloatingTargetPlatformVersion)' != 'true' And '$(BundleOriginalResources)' != 'false'))\">");
+		} else {
+			writer.WriteLine ($"	<ImportGroup Condition=\" '$(TargetPlatformIdentifier)' == '{platform}' And '$(UsingAppleNETSdk)' != 'true' And $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '{tfv}')) And '$(TargetPlatformVersion)' == '{tpv}'\">");
+		}
 		writer.WriteLine ($"		<Import Project=\"Sdk.props\" Sdk=\"Microsoft.{platform}.Sdk.{workloadVersion}\" />");
 		if (hasWindows)
 			writer.WriteLine ($"		<Import Project=\"Sdk.props\" Sdk=\"Microsoft.{platform}.Windows.Sdk.Aliased.{tfm}\" Condition=\" $([MSBuild]::IsOSPlatform('windows'))\" />");

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -65,6 +65,7 @@ test.config: Makefile $(TOP)/Make.config $(TOP)/mk/mono.mk $(TOP)/eng/Version.De
 	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),$(platform)_NUGET_RUNTIME_MANAGED_NAME=$($(platform)_NUGET_RUNTIME_MANAGED_NAME)\\n)" | sed 's/^ //' >> $@
 	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),$(foreach rid,$(DOTNET_$(platform)_RUNTIME_IDENTIFIERS),$(rid)_NUGET_RUNTIME_NAME=$($(rid)_NUGET_RUNTIME_NAME)\\n))" | sed 's/^ //' >> $@
 	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),SUPPORTED_API_VERSIONS_$(platform)='$(SUPPORTED_API_VERSIONS_$(platform))'\\n)" | sed 's/^ //' >> $@
+	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),$(platform)_TARGET_PLATFORM_VERSION_LIBRARY=$($(platform)_TARGET_PLATFORM_VERSION_LIBRARY)\\n)" | sed 's/^ //' >> $@
 	@printf "ENABLE_XAMARIN=$(ENABLE_XAMARIN)\n" >> $@
 	@printf "ENABLE_ADR=$(ENABLE_ADR)\n" >> $@
 	@printf "XCODE_IS_STABLE=$(XCODE_IS_STABLE)\n" >> $@
@@ -97,6 +98,7 @@ test-system.config: Makefile $(TOP)/Make.config $(TOP)/mk/mono.mk $(TOP)/eng/Ver
 	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),$(platform)_NUGET_RUNTIME_MANAGED_NAME=$($(platform)_NUGET_RUNTIME_MANAGED_NAME)\\n)" | sed 's/^ //' >> $@
 	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),$(foreach rid,$(DOTNET_$(platform)_RUNTIME_IDENTIFIERS),$(rid)_NUGET_RUNTIME_NAME=$($(rid)_NUGET_RUNTIME_NAME)\\n))" | sed 's/^ //' >> $@
 	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),SUPPORTED_API_VERSIONS_$(platform)='$(SUPPORTED_API_VERSIONS_$(platform))'\\n)" | sed 's/^ //' >> $@
+	@printf "$(foreach platform,$(DOTNET_PLATFORMS_UPPERCASE),$(platform)_TARGET_PLATFORM_VERSION_LIBRARY=$($(platform)_TARGET_PLATFORM_VERSION_LIBRARY)\\n)" | sed 's/^ //' >> $@
 	@printf "ENABLE_XAMARIN=$(ENABLE_XAMARIN)\n" >> $@
 	@printf "ENABLE_ADR=$(ENABLE_ADR)\n" >> $@
 	@printf "XCODE_IS_STABLE=$(XCODE_IS_STABLE)\n" >> $@

--- a/tests/common/Configuration.cs
+++ b/tests/common/Configuration.cs
@@ -180,6 +180,7 @@ namespace Xamarin.Tests {
 			ParseConfigFiles (FindConfigFiles ("configure.inc"));
 			ParseConfigFiles (FindConfigFiles ("Make.config.local"));
 			ParseConfigFiles (FindConfigFiles ("Make.config"));
+			ParseConfigFiles (FindConfigFiles ("Make.versions"));
 		}
 
 		static void ParseConfigFiles (IEnumerable<string> files)

--- a/tests/dotnet/MyClassLibrary/MacCatalyst/MyClass.cs
+++ b/tests/dotnet/MyClassLibrary/MacCatalyst/MyClass.cs
@@ -1,8 +1,0 @@
-using System;
-namespace MyClassLibrary {
-	public class MyClass {
-		public MyClass ()
-		{
-		}
-	}
-}

--- a/tests/dotnet/MyClassLibrary/MacCatalyst/MyClassLibrary.csproj
+++ b/tests/dotnet/MyClassLibrary/MacCatalyst/MyClassLibrary.csproj
@@ -3,5 +3,6 @@
 	<PropertyGroup>
 		<TargetFramework>net$(BundledNETCoreAppTargetFrameworkVersion)-maccatalyst</TargetFramework>
 	</PropertyGroup>
+	<Import Project="..\shared.csproj" />
 </Project>
 

--- a/tests/dotnet/MyClassLibrary/MyClass.cs
+++ b/tests/dotnet/MyClassLibrary/MyClass.cs
@@ -1,6 +1,9 @@
 using System;
+
+using Foundation;
+
 namespace MyClassLibrary {
-	public class MyClass {
+	public class MyClass : NSObject {
 		public MyClass ()
 		{
 		}

--- a/tests/dotnet/MyClassLibrary/iOS/MyClass.cs
+++ b/tests/dotnet/MyClassLibrary/iOS/MyClass.cs
@@ -1,8 +1,0 @@
-using System;
-namespace MyClassLibrary {
-	public class MyClass {
-		public MyClass ()
-		{
-		}
-	}
-}

--- a/tests/dotnet/MyClassLibrary/iOS/MyClassLibrary.csproj
+++ b/tests/dotnet/MyClassLibrary/iOS/MyClassLibrary.csproj
@@ -3,5 +3,6 @@
 	<PropertyGroup>
 		<TargetFramework>net$(BundledNETCoreAppTargetFrameworkVersion)-ios</TargetFramework>
 	</PropertyGroup>
+	<Import Project="..\shared.csproj" />
 </Project>
 

--- a/tests/dotnet/MyClassLibrary/macOS/MyClass.cs
+++ b/tests/dotnet/MyClassLibrary/macOS/MyClass.cs
@@ -1,8 +1,0 @@
-using System;
-namespace MyClassLibrary {
-	public class MyClass {
-		public MyClass ()
-		{
-		}
-	}
-}

--- a/tests/dotnet/MyClassLibrary/macOS/MyClassLibrary.csproj
+++ b/tests/dotnet/MyClassLibrary/macOS/MyClassLibrary.csproj
@@ -3,5 +3,6 @@
 	<PropertyGroup>
 		<TargetFramework>net$(BundledNETCoreAppTargetFrameworkVersion)-macos</TargetFramework>
 	</PropertyGroup>
+	<Import Project="..\shared.csproj" />
 </Project>
 

--- a/tests/dotnet/MyClassLibrary/shared.csproj
+++ b/tests/dotnet/MyClassLibrary/shared.csproj
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+	<ItemGroup>
+		<Compile Include="../*.cs" />
+	</ItemGroup>
+</Project>

--- a/tests/dotnet/MyClassLibrary/tvOS/MyClass.cs
+++ b/tests/dotnet/MyClassLibrary/tvOS/MyClass.cs
@@ -1,8 +1,0 @@
-using System;
-namespace MyClassLibrary {
-	public class MyClass {
-		public MyClass ()
-		{
-		}
-	}
-}

--- a/tests/dotnet/MyClassLibrary/tvOS/MyClassLibrary.csproj
+++ b/tests/dotnet/MyClassLibrary/tvOS/MyClassLibrary.csproj
@@ -3,5 +3,6 @@
 	<PropertyGroup>
 		<TargetFramework>net$(BundledNETCoreAppTargetFrameworkVersion)-tvos</TargetFramework>
 	</PropertyGroup>
+	<Import Project="..\shared.csproj" />
 </Project>
 

--- a/tests/dotnet/UnitTests/ProjectTest.cs
+++ b/tests/dotnet/UnitTests/ProjectTest.cs
@@ -90,18 +90,33 @@ namespace Xamarin.Tests {
 			Assert.AreEqual ("3.14", infoPlist.GetString ("CFBundleShortVersionString").Value, "CFBundleShortVersionString");
 		}
 
-		[TestCase ("iOS")]
-		[TestCase ("tvOS")]
-		[TestCase ("macOS")]
-		[TestCase ("MacCatalyst")]
+		[TestCase (ApplePlatform.iOS)]
+		[TestCase (ApplePlatform.TVOS)]
+		[TestCase (ApplePlatform.MacOSX)]
+		[TestCase (ApplePlatform.MacCatalyst)]
 		[Category ("WindowsInclusive")]
-		public void BuildMyClassLibrary (string platform)
+		public void BuildMyClassLibrary (ApplePlatform platform)
 		{
 			Configuration.IgnoreIfIgnoredPlatform (platform);
-			var project_path = GetProjectPath ("MyClassLibrary", platform);
+			var configuration = "Debug";
+			var project = "MyClassLibrary";
+			var project_path = GetProjectPath (project, platform: platform);
 			Clean (project_path);
-			var result = DotNet.AssertBuild (project_path, verbosity);
+			var properties = GetDefaultProperties ();
+			var result = DotNet.AssertBuild (project_path, properties);
 			Assert.That (result.StandardOutput.ToString (), Does.Not.Contain ("Task \"ILLink\""), "Linker executed unexpectedly.");
+
+			var expectedReferenceVersionString = Configuration.GetVariable ($"{platform.AsString ().ToUpperInvariant ()}_TARGET_PLATFORM_VERSION_LIBRARY", "");
+			Assert.That (expectedReferenceVersionString, Is.Not.EqualTo (""), $"Expected to find a default TPV for libraries; this is a problem in the test setup.");
+
+			var dll = Path.Combine (Path.GetDirectoryName (project_path)!, "bin", configuration, $"{Configuration.DotNetTfm}-{platform.AsString ()}", project + ".dll");
+			Assert.That (dll, Does.Exist, "Exists");
+			using var ad = AssemblyDefinition.ReadAssembly (dll, new ReaderParameters { ReadingMode = ReadingMode.Deferred });
+			var r = ad.MainModule.AssemblyReferences.Where (v => v.Name == $"Microsoft.{platform.AsString ()}").First ();
+			var actualReferenceVersionString = $"{r.Version.Major}.{r.Version.Minor}";
+			Assert.AreEqual (expectedReferenceVersionString, actualReferenceVersionString, $"Referenced version of Microsoft.{platform.AsString ()}.dll");
+			Assert.That (r.Version.Build, Is.EqualTo (0), "Build");
+			Assert.That (r.Version.Revision, Is.EqualTo (0), "Revision");
 		}
 
 		[TestCase (ApplePlatform.iOS)]

--- a/tests/xharness/Harness.cs
+++ b/tests/xharness/Harness.cs
@@ -568,7 +568,8 @@ namespace Xharness {
 			return FindConfigFiles (useSystemXamarinIOSMac ? "test-system.config" : "test.config")
 				.Concat (FindConfigFiles ("configure.inc"))
 				.Concat (FindConfigFiles ("Make.config"))
-				.Concat (FindConfigFiles ("Make.config.local"));
+				.Concat (FindConfigFiles ("Make.config.local"))
+				.Concat (FindConfigFiles ("Make.versions"));
 		}
 
 		void ParseConfigFile (string file, Dictionary<string, string> configuration)

